spec/requests/api/pages_spec.rb

```diff
@@ -0,0 +1,116 @@
+require_relative 'shared_language_pages.rb'
+require 'rails_helper'
+
+describe "api/pages" do
+
+  def json
+    JSON.parse(response.body)
+  end
+
+
+  before :each do
+    # I'm rounding the time. Ruby deals with time in nanoseconds whereas the database deals with time in microsecond
+    # precision. If I don't round the time, the expectation comparing the JSON response expects data from the DB
+    # with nanosecond precision.
+    @time_now = Time.at(Time.now.to_i)
+    allow(Time).to receive(:now).and_return(@time_now)
+  end
+
+  describe 'GET pages' do
+    context 'with no specified language' do
+      let!(:featured_pages) { create_list :page, 50, featured: true }
+      let!(:mvp_pages) { create_list :page, 50, featured: false }
+      let!(:last_featured_page) { create :page, title: 'I am the latest featured page', featured: true, slug: 'garden_slug' }
+      let!(:last_mvp_page) { create :page, title: 'I am the latest test page', featured: false}
+
+      it 'gets a hundred of both featured and unfeatured pages in a reversed order if requested without an id' do
+        get api_pages_path
+        expect(response).to be_success
+        # Includes both featured and unfeatured pages.
+        expect(json).to include last_featured_page.as_json
+        expect(json).to include last_mvp_page.as_json
+        # Limits its reach to the latest hundred pages if there are more than a hundred pages to search through.
+        expect(json).to match Page.last(100).reverse.as_json
+      end
+
+      it 'gets a single page if searched by an id of a page that exists' do
+        get api_pages_path(:id => last_mvp_page.id.to_s)
```

Comment:
I'm being finicky (apologies), but can you kill the hash rockets (in the name of style consistency).